pkg/search/postgres/select_query_test.go

```diff
@@ -0,0 +1,309 @@
+//go:build sql_integration
+
+package postgres_test
+
+import (
+	"context"
+	"testing"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	pkgPG "github.com/stackrox/rox/pkg/search/postgres"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stackrox/rox/tools/generate-helpers/pg-table-bindings/multitest/postgres"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Struct1 struct {
+	TestString string `db:"teststring"`
+}
+
+type Struct2 struct {
+	TestNestedString string `db:"testnestedstring"`
+	TestNestedBool   bool   `db:"testnestedbool"`
+}
+
+type Struct2GrpBy1 struct {
+	TestNestedString []string `db:"testnestedstring"`
+	TestNestedBool   []bool   `db:"testnestedbool"`
+	TestString       string   `db:"teststring"`
+}
+
+type Struct2GrpBy2 struct {
+	TestNestedString []string `db:"testnestedstring"`
+	TestNestedBool   []bool   `db:"testnestedbool"`
+	TestString       string   `db:"teststring"`
+	TestBool         bool     `db:"testbool"`
+}
+
+type Struct3 struct {
+	TestString       string `db:"teststring"`
+	TestNestedString string `db:"testnestedstring"`
+}
+
+func TestSelectQueryResults(t *testing.T) {
+	t.Parallel()
+
+	ctx := sac.WithAllAccess(context.Background())
+	testDB := pgtest.ForT(t)
+	defer testDB.Teardown(t)
+
+	store := postgres.New(testDB.Pool)
+
+	testStructs := getTestStructs()
+
+	for _, s := range testStructs {
+		require.NoError(t, store.Upsert(ctx, s))
+	}
+
+	for _, c := range []struct {
+		desc           string
+		q              *v1.Query
+		resultStruct   any
+		expectedError  string
+		expectedQuery  string
+		expectedResult any
+	}{
+		{
+			desc: "base schema; no select",
+			q: search.NewQueryBuilder().
+				AddExactMatches(search.TestString, "acs").ProtoQuery(),
+			resultStruct:   Struct1{},
+			expectedError:  "select portion of the query cannot be empty",
+			expectedResult: nil,
+		},
+		{
+			desc: "base schema; select",
+			q: search.NewQueryBuilder().
+				AddSelectFields(search.TestString).ProtoQuery(),
+			resultStruct:  Struct1{},
+			expectedQuery: "select test_multi_key_structs.String_ teststring from test_multi_key_structs",
+			expectedResult: []*Struct1{
+				{"acs"},
+				{"acs"},
+				{"bcs"},
+				{"bcs"},
+			},
+		},
+		{
+			desc: "base schema; select w/ where",
+			q: search.NewQueryBuilder().
+				AddSelectFields(search.TestString).
+				AddExactMatches(search.TestString, "acs").ProtoQuery(),
+			resultStruct:  Struct1{},
+			expectedQuery: "select test_multi_key_structs.String_ teststring from test_multi_key_structs where test_multi_key_structs.String_ = $1",
```

Comment:
Curious why .String_ vs .Nested (without the _). Should be consistent right?

Reply:
not following. `string_` is the table column name.